x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/action/DeleteDataStreamTransportAction.java

```diff
@@ -106,9 +108,11 @@ static ClusterState removeDataStream(
         ClusterState currentState,
         DeleteDataStreamAction.Request request
     ) {
-        Set<String> dataStreams = new HashSet<>(
-            indexNameExpressionResolver.dataStreamNames(currentState, request.indicesOptions(), request.getNames())
-        );
+        IndicesOptions options = request.indicesOptions();
```

Comment:
I don't fully understand the purpose of this logic here. It always ensures that expressions get expanded to open data streams / indices. Why is this needed? 

Also the rest spec for this and get data stream api only indicates that `expand_wildcards` is supported, but I think the way the corresponding request class and with this logic also the other indices options rest parameters are supported as well? Is the intention of this logic to only support `expand_wildcards` in get and delete data streams apis?

Reply:
Yes, the intention was to only support `expand_wildcards` but to avoid duplicating parsing parameters code.
`open`/`closed` has no meaning in terms of data stream (we can't close data stream, right?), but if we specify only `hidden` no data stream will be returned (that's the way `IndexNameExpressionResolver` works). This is just convenience for a user.
Other options should be ignored as well by the get/delete data stream APIs so no harm with parsing them.

Reply:
Ok, I see. Can the logic that rewrites the indices options be extracted in a static helper method with a comment describing what you just explained?

Reply:
Yes, added